lib/dolly/connection.rb

```diff
@@ -75,10 +76,15 @@ def request(method, resource, data = {})
     def start_request(req)
       Net::HTTP.start(req.uri.hostname, req.uri.port) do |http|
         req.basic_auth env['username'], env['password'] if env['username']&.present?
+        http.use_ssl = secure?
         http.request(req)
       end
     end
 
+    def secure?
+      env['protocol'] == SECURE_PROTOCOL
```

Comment:
Just wondering if we should drop the protocol part of this since it doesn't look like we use that in the underlying connection and instead just use a `use_ssl: true` type flag in the config.

Reply:
`protocol` is used on https://github.com/amco/dolly/blob/3.0/lib/dolly/configuration.rb#L13 to set the `base_uri` for the connection.

Reply:
Right, but it could be hard coded as constants and the correct one used based on the use_ssl flag. It is fine either way, just seems we could create one less typo issue and not have to worry about a string comparison. Was just a thought.

Reply:
ok, will raise an issue to do so

Reply:
I just need to check as this will make current config invalid, as not many projects have Dolly3, should be fine, but there are some on development already using it.